app/src/main/java/ar/com/wolox/android/example/ui/example/login/LoginPresenter.java

```diff
@@ -11,4 +12,39 @@
 
     @Inject
     LoginPresenter() {}
-}
\ No newline at end of file
+
+    void onLoginValidation(String email, String password) {
+        boolean isValidEmail = validateEmail(email);
+        boolean isValidPassword = validatePassword(password);
+
+        if (isValidEmail && isValidPassword) {
+            getView().onValidatedLogin();
+        }
+    }
+
+    private boolean validateEmail(String email) {
+        boolean isValidEmail = false;
+
+        if (TextUtils.isEmpty(email)) {
+            getView().onEmptyEmail();
```

Comment:
I would prefer that in this line appears 'isValidEmail = false' and then in onLoginValidation you decide where to go

Reply:
✔ **Discussed in morning call:** We agreed to keep it in the way it is